### ISSUE TITLE
Session Desktop fails to launch

### ISSUE THREAD

**Issue by beantaco** [oxen-io/session-desktop#1633]
After updating Session Desktop to 1.6.1, I can no longer launch it.
I'm running a Debian related OS on amd64.

When I launch it from a terminal, I get the following:

~~~
$ session-desktop
/usr/bin/session-desktop: line 14: /usr/bin/session-desktop-bin: No such file or directory
~~~

The script `/usr/bin/session-desktop` is getting the wrong directory path `/usr/bin` (line 13) and attaching it (line 14) to the executable name `session-desktop-bin` which is located in `/opt/Session`.

I have the following relevant executable related files installed:
(Note: `/opt/Session/chrome-sandbox` doesn't exist)

~~~
$ ls -l /bin/session-desktop* /usr/bin/session-desktop* /opt/Session/session-desktop* /opt/Session/chrome-sandbox
ls: cannot access '/opt/Session/chrome-sandbox': No such file or directory
lrwxrwxrwx ..        28 .. /bin/session-desktop -> /opt/Session/session-desktop
-rwxr-xr-x ..       733 .. /opt/Session/session-desktop
-rwxr-xr-x .. 116332248 .. /opt/Session/session-desktop-bin
lrwxrwxrwx ..        28 .. /usr/bin/session-desktop -> /opt/Session/session-desktop
~~~

I can launch Session Desktop by directly calling `/opt/Session/session-desktop-bin --no-sandbox` with no apparent problems. (Aside: The `--no-sandbox` flag seems to be a workaround that disables some kind of sandbox for some reason... Does this pose a security risk?)

The following information may be unhelpful, but when installing version 1.6.1 I got the following message about `/opt/Session/chrome-sandbox`:

~~~
chmod: cannot access '/opt/Session/chrome-sandbox': No such file or directory
~~~

**Comment by Bilb** [comment]
@beantaco could you give a try to 1.6.1 or more recent and let me know if your issue is fixed already?

**Comment by beantaco** [comment]
I tried with versions 1.6.2 and 1.6.4 but the issue persists.

**Comment by Bilb** [comment]
Can you tell what is your debian related os and version so I can debug that?

**Comment by beantaco** [comment]
I am using Debian buster with all packages updated.

I tried uninstalling and installing Session again, but the issue persists.

To clarify, Session does launch by using the provided `session-desktop.desktop` file (which references `/opt/Session/session-desktop`) but not by invoking `session-desktop` on the command line (which executes `/usr/bin/session-desktop`, a symlink to `/opt/Session/session-desktop`).

My suspicion is the problem lies in using the symlink `/usr/bin/session-desktop` to execute `/opt/Session/session-desktop`. This is a bash script that attempts to set `SCRIPT_DIR` (line 13) by getting the directory path of the current script. When `/usr/bin/session-desktop` is executed, `SCRIPT_DIR` is set to `/usr/bin` and not `/opt/Session`, giving rise to the error that `/usr/bin/session-desktop-bin` could not be found (line 14).

Did `/usr/bin/session-desktop-bin` exist in previous versions of Session? Or was `SCRIPT_DIR` hard-coded to `/opt/Session` in previous versions?

**Comment by peepopoggers** [comment]
any updates?

**Comment by Bilb** [comment]
Could you give 1.6.10 a try?

**Comment by beantaco** [comment]
It appears to be fixed for version 1.6.10. :-)

**Comment by Bilb** [comment]
nice !
